### PR TITLE
docs: update docs prior 1.11.0 release

### LINKED ISSUE #3212: Update User management documentation
Once PR 
- https://github.com/argilla-io/argilla/pull/3169
- https://github.com/argilla-io/argilla/pull/3180
- https://github.com/argilla-io/argilla/pull/3104

are done, we need to review the docs referring to the users and workspaces management, since several changes must be included

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

This PR contains the following linked PRs to update the documentation before releasing Argilla 1.11.0:
* #3236 
* #3237 

**Type of change**

- [X] Documentation update

### CODE REVIEW

**Review by davidberenstein1957 — APPROVED** [review]

File: docs/_source/getting_started/installation/configurations/workspace_management.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+### List all the existing `Workspaces`
+
+#### Python client
+
+You can also list all the existing workspaces in Argilla using the `list` method. It will return a list of `Workspace` instances.
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/workspace_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/workspace_management.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+### Create a new `Workspace`
+
+#### Python client
+
+Creating a workspace in Argilla is now as easy as calling the `create` method from the `Workspace` class. It will return a `Workspace` instance.
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/workspace_management.md]:
```suggestion
rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/workspace_management.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+### Get a `Workspace` by name
+
+#### Python client
+
+You can get a workspace by its name using the `from_name` method. It must exist in advance in Argilla, otherwise an exception will be raised.
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/workspace_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/workspace_management.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+Additionally, if you know the `id` of the workspace, you can get it directly using the `from_id` method. It must exist in advance in Argilla, otherwise an exception will be raised.
+
+:::{note}
+The `id` of a workspace is a UUID, and it is generated automatically when you create a new workspace.
+:::
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/workspace_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/workspace_management.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+Once you instantiate a `Workspace` instance from a workspace in Argilla, you can add, list, or delete users from it. But note that just the `owner` has sufficient permissions to perform those operations.
+
+:::{note}
+As of the 1.11.0 version of the Python client, to add and delete users from a `Workspace` one must use the `UUID` which is the unique identifier in Argilla for each user. To do so, one can simply use the `id` attribute of the `User` class. More information about it can be found in the [User - Python Reference](../reference/python/python_users.rst) or in the [User Management](user_management.md) guide.
+:::
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/workspace_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/user_management.md [hunk 6]
```diff
@@ -181,50 +195,79 @@ Options:
...
 ```
 
+#### Python client
+
+You can also create a new user in Argilla using the `create` classmethod in the `User` class.
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/user_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/user_management.md [hunk 7]
```diff
@@ -181,50 +195,79 @@ Options:
...
-#### Creating an annotator user assigned to a workspace
+#### Python client
 
-**CLI:**
+You can get a user by id using the `from_id` classmethod in the `User` class.
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/user_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/user_management.md [hunk 8]
```diff
@@ -242,115 +285,65 @@ User successfully created:
...
-   "workspaces":[
-      "ws2"
-   ],
-   "api_key":"SrX9T_4DAWK65Ztp4sADfB3g05t3bpwjwgfIwR3BP90uRg_LkWlsBXccAI9KTRbedxMNDdw15pM",
-   "inserted_at":"2023-03-16T11:17:35.462774",
-   "updated_at":"2023-03-16T11:17:35.462774"
-}
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/user_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/user_management.md [hunk 9]
```diff
@@ -396,99 +389,24 @@ Migrating User with username 'daisy'
...
-         "latam"
-      ],
-      "api_key":"a8168929-8668-494c-b7a5-98cd35740d9b",
-      "inserted_at":"2023-03-16T11:31:12.990718",
-      "updated_at":"2023-03-16T11:31:12.990718"
-   }
-]
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/user_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

File: docs/_source/getting_started/installation/configurations/user_management.md [hunk 10]
```diff
@@ -150,7 +117,54 @@ Path: /path/to/alembic/versions/1769ee58fbb4_create_workspaces_users_table.py
...
+```
+
+#### Python client
+
+You can get the current active user in Argilla using the `me` classmethod in the `User` class. Note that the `me` method will return the active user as specified via the credentials provided via `rg.init`.
+
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<API_KEY>")
```
Comment (davidberenstein1957) [inline docs/_source/getting_started/installation/configurations/user_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

**Review by davidberenstein1957 — APPROVED** [review]

**Comment by davidberenstein1957** [comment]
@alvarobartt LFTM, I only opt for `rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")` to align with our env vars.

File: docs/_source/getting_started/installation/configurations/workspace_management.md (hunk 2 above)
Comment (alvarobartt) [reply docs/_source/getting_started/installation/configurations/workspace_management.md]:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

**Comment by alvarobartt** [comment]
cc @davidberenstein1957 it seems that the Python ref for `FeedbackDataset` is still failing, can you please check? Thanks 🤗

**Review by davidberenstein1957 — APPROVED** [review]

**Pull request merged**